HostLibraryTests/hip/RunGEMMKernelTileSelection_test.cpp

```diff
@@ -83,15 +83,10 @@ struct RunGEMMKernelSolutionSelectionTest : public ::testing::TestWithParam<Cont
 
         std::mt19937 rng;
 
-        InitTensor(a_h.data(), problem.a(), RandomInt<float>(rng));
-        InitTensor(b_h.data(), problem.b(), RandomAlternatingInt<float>(rng));
-        InitTensor(c_h.data(), problem.c(), RandomInt<float>(rng));
-        InitTensor(d_in_h.data(), problem.d(), RandomInt<float>(rng));
-
-        // InitTensor(a_h.data(), problem.a, Iota<float>());
-        // InitTensor(b_h.data(), problem.b, Iota<float>());
-        // InitTensor(c_h.data(), problem.c, RandomInt<float>());
-        // InitTensor(d_h.data(), problem.d, RandomInt<float>());
+        InitTensor(a_h.data(), problem.a(), RandomInt<float>(), rng);
```

Comment:
The reason this file was not being included in the tests is because the libraries were added to the regular `RunGEMMKernel_test.cpp` file, since the interface doesn't vary between tile-aware and distance metrics.

`RunGEMMKernel_test.cpp` has also been improved significantly with results caching, bounds checking, and support for different data types.  We should delete this file and add the new libraries to the `RunGEMMKernel_test.cpp`.